panther_analysis_tool/detection.py

```diff
@@ -189,3 +249,597 @@ def get_module(  # pylint: disable=arguments-differ
+        """
+        has_raw_code = bool(config.get("body"))
+
+        importer: Optional[BaseImporter] = None
+        if has_raw_code:
+            resource = config["body"]
+            importer = RawStringImporter(tmp_dir=_DETECTION_FOLDER)
+        else:
+            resource = config["path"]
+            importer = FilesystemImporter()
+
+        return importer.get_module(identifier, resource)
+
+    def _check_defined_functions(self) -> Dict[str, bool]:
+        function_definitions = {}
+        for name in AUXILIARY_FUNCTIONS:
+            function_definitions[name] = self._is_function_defined(name)
+        return function_definitions
+
+    @property
+    def matcher_function(self) -> str:
+        # the matcher function will differ depending on detection type
+        return "rule"
+
+    @property
+    def trigger_alert(self) -> bool:
+        # the returned value from the detection will trigger an alert
+        # when a rule returns true and when a policy return false
+        return True
+
+    @property
+    def module(self) -> Any:
+        """Used to expose the loaded python module to the engine,
+        solely added in to support unit test mocking"""
+        return self._module
+
+    @property
+    def setup_exception(self) -> Any:
+        """Used to expose the setup exception to the engine"""
+        return self._setup_exception
+
+    @setup_exception.setter
+    def setup_exception(self, val: Any) -> Any:
+        """Used by the engine to set _setup_exception"""
+        self._setup_exception = val
+
+    def run(
+        self, event: PantherEvent, outputs: dict, outputs_names: dict, batch_mode: bool = True
+    ) -> DetectionResult:
+        """
+        Analyze a log line with this rule and return True, False, or an error.
+        :param event: The event to run the rule against
+        :param outputs: Destinations loaded from the panther-outputs-api
+        :param outputs_names: Destinations mapped by their display name
+        :param batch_mode: Whether the rule runs as part of the log analysis
+        or as part of a simple rule test. In batch mode, title/dedup functions
+        are not checked if the rule won't trigger an alert and also title()/dedup()
+        won't raise exceptions, so that an alert won't be missed.
+        """
+        detection_result = DetectionResult(
+            detection_id=self.detection_id, detection_severity=self.detection_severity
+        )
+        # If there was an error setting up the rule
+        # return early
+        if self._setup_exception:
+            detection_result.setup_exception = self._setup_exception
+            return detection_result
+
+        try:
+            detection_result.matched = self._run_detection(event)
+        except Exception as err:  # pylint: disable=broad-except
+            detection_result.detection_exception = err
+
+        if batch_mode and detection_result.matched != self.trigger_alert:
```

Comment:
`matched` may have a misleading meaning for policies? You could read it like `matched -> False` means _does not match the compliance requirements_ , though.

Reply:
Truth value checking can be a bit tricky here. Although the equality comparison also works, since booleans are singletons, you can use the object identity operator `is` and maybe consider renaming the property, something like:

```python
detection_result.matched is self.matcher_alert_value
```

Reply:
> You could read it like matched -> False means does not match the compliance requirements , though.

I'd say if a resource doesn't match the policy (`mached == False`) then it isn't compliant, so I could make an argument it makes sense in that case too 🤷🏻‍♀️

Reply:
Agreed, that's what I meant in my comment actually!